mllib/src/main/scala/org/apache/spark/ml/fpm/FPGrowth.scala

```diff
@@ -361,20 +387,22 @@ private[fpm] object AssociationRules {
         dataset: Dataset[_],
         itemsCol: String,
         freqCol: String,
-        minConfidence: Double): DataFrame = {
+        minConfidence: Double,
+        itemSupport: Map[Any, Long]): DataFrame = {
```

Comment:
Shouldn't this be a `Map[T, Long]`? it's required to be anyway below. It's worth adding to the scaladoc too.